resources/Microsoft.VSCode.DSC/Microsoft.VSCode.DSC.psm1

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum Ensure
+{
+    Absent
+    Present
+}
+
+#region DSCResources
+[DSCResource()]
+class VSCodeExtension
+{
+    [DscProperty(Key)]
+    [string] $Name
+
+    [DscProperty()]
+    [string] $Version
+
+    [DscProperty()]
+    [Ensure] $Ensure = [Ensure]::Present
+
+    [DscProperty(NotConfigurable)]
+    [hashtable] $InstalledExtensions = @{}
```

Comment:
Why do you need this as a member? You are only in Get and it seems that you just iterate over then to find it.

Reply:
I wanted to have a way to show the users what extensions are currently installed along with their versions. I made it NotConfigurable so they can't modify it and it is only to be used as a reference. I figured more info is better than none even if it is not absolutely necessary.

Reply:
This is export functionality, which we don't have right now but may in the future.  Remove the property.

Reply:
Removed property